syntax = "proto2";

package package.name;

service Greeter {
  rpc Transmitter (Request) returns (Response.Galaxy) {}
}

message Response {

        enum Alignment {
        Ally = 0;
        Enemy = 1;
      }

        enum Class {
        Corvette = 0;
        Frigate = 1;
        Cruiser = 2;
        Destroyer = 3;
        Carrier = 4;
        Dreadnought = 5;
      }

    message Ship {

        message Officer {
        required string first_name = 1;
        required string last_name = 2;
        required string rank = 3;
      }

        required string name = 1;
        required int32 crew_size = 2;
        required float length = 3;
        required bool armed = 4;

        required Alignment alignment = 5;
        required Class class_ = 6;

        repeated Officer officers = 7;
    }
    message Galaxy {
        repeated Ship ships = 1; 
    }
}

message Request {
    optional string coordinates = 1;
}